src/chrome/breakOnLoadHelper.ts

```diff
@@ -157,14 +157,31 @@ export class BreakOnLoadHelper {
         const hitBreakpoints = notification.hitBreakpoints;
         let allStopOnEntryBreakpoints = true;
 
+        const pausedScriptId = notification.callFrames[0].location.scriptId;
+        const pausedScriptUrl = this._chromeDebugAdapter.scriptsById.get(pausedScriptId).url;
+        const mappedUrl = await this._chromeDebugAdapter.pathTransformer.scriptParsed(pausedScriptUrl);
+
         // If there is a breakpoint which is not a stopOnEntry breakpoint, we appear as if we hit that one
         // This is particularly done for cases when we end up with a user breakpoint and a stopOnEntry breakpoint on the same line
-        hitBreakpoints.forEach(bp => {
-            if (!this._stopOnEntryBreakpointIdToRequestedFileName.has(bp)) {
+        for (let bp of hitBreakpoints) {
+            let regexAndFileNames = this._stopOnEntryBreakpointIdToRequestedFileName.get(bp);
+            if (!regexAndFileNames) {
                 notification.hitBreakpoints = [bp];
                 allStopOnEntryBreakpoints = false;
+            } else {
+                const normalizedMappedUrl = this._chromeDebugAdapter.fixPathCasing(mappedUrl);
+                if (regexAndFileNames[1].has(normalizedMappedUrl)) {

```

Comment:
We should really change this to an object with named properties, not an array. Don't know why I didn't make that comment earlier

Reply:
And this is for keeping track of when multiple files are covered by a single break on load breakpoint, right?

Reply:
Awesome feedback. It made me realized that I was missing a bunch of code :)